regression_tests/leader_service.go

```diff
@@ -0,0 +1,163 @@
+// creating and managing actors, and transferring funds within the wallet ecosystem.
+type LeaderService struct {
+	cfg      *LeaderServiceConfig // Configuration for the service.
+	adminAPI *wallet.AdminAPI     // Admin API client for privileged operations.
+	userAPI  *wallet.UserAPI      // User API client for user-level operations.
+	actors   []*Actor             // List of actors created by this service.
+	domain   string               // Paymail domain used by the service.
+}
+
+// Paymail returns the paymail address of the leader.
+// The paymail is constructed as "Leader@" followed by the configured domain.
+func (l *LeaderService) Paymail() string {
+	return "Leader@" + l.domain
+}
+
+// Balance retrieves the current satoshi balance for the leader account.
+// It accepts a context parameter to manage cancellation and timeouts.
+// The method fetches the xPub key of the user account and uses it to determine the current balance.
+// On success, it returns the balance in satoshis and a nil error. If the API call fails, it returns a non-nil error.
+func (l *LeaderService) Balance(ctx context.Context) (uint64, error) {
+	xpub, err := l.userAPI.XPub(ctx)
+	if err != nil {
+		return 0, fmt.Errorf("failed to fetch xPub for balance retrieval: %w", err)
+	}
+	return xpub.CurrentBalance, nil
+}
+
+// Transactions retrieves the list of transactions associated with the leader's account.
+// It accepts a context parameter to manage cancellation and timeouts.
+// On success, it returns a slice of transactions contained in the paginated response and a nil error.
+// If the API call fails, it returns a non-nil error.
+func (l *LeaderService) Transactions(ctx context.Context) ([]*response.Transaction, error) {
+	page, err := l.userAPI.Transactions(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch transactions: %w", err)
+	}
+	return page.Content, nil
+}
+
+// TransferFunds transfers a specified amount of satoshis to the recipient's paymail address.
+// It accepts a context parameter to manage cancellation and timeouts.
+// On success, it returns the transaction representing the fund transfer and a nil error.
+// If the API call fails, it returns a non-nil error.
+func (l *LeaderService) TransferFunds(ctx context.Context, paymail string, funds uint64) (*response.Transaction, error) {
+	balance, err := l.Balance(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch balance: %w", err)
+	}
+	if balance < funds {
+		return nil, fmt.Errorf("insufficient balance: %d available, %d required", balance, funds)
+	}
+
+	recipient := commands.Recipients{To: paymail, Satoshis: funds}
+	transaction, err := l.userAPI.SendToRecipients(ctx, &commands.SendToRecipients{
+		Recipients: []*commands.Recipients{&recipient},
+		Metadata:   map[string]any{"description": "regression-test"},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to transfer funds to %s: %w", paymail, err)
+	}
+	return transaction, nil
```

Comment:
Looks almost exactly like `TransferFunds` for `actor`. Consider make those two "classes" more DRY.